src/Microsoft.AspNetCore.Mvc.Testing/WebHostBuilderExtensions.cs

```diff
@@ -0,0 +1,112 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Mvc.ApplicationParts;
+using Microsoft.AspNetCore.Mvc.Internal;
+using Microsoft.AspNetCore.Mvc.Testing.Internal;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.AspNetCore.Mvc.Testing
+{
+    /// <summary>
+    /// Testing extensions for creating an configuring an <see cref="IWebHostBuilder"/> for end to end in memory tests.
+    /// </summary>
+    public static class WebHostBuilderTestingExtensions
+    {
+        private static readonly string CreateDefaultBuilder = nameof(CreateDefaultBuilder);
+
+        /// <summary>
+        /// Initializes a new <see cref="IWebHostBuilder"/> instance by calling a method 
+        /// <c>public static CreateDefaultBuilder(string [] args)</c> on the
+        /// entry point class for the assembly. Tipically, <c>Program</c>.
+        /// </summary>
+        /// <typeparam name="TStartup">The type of the Startup class.</typeparam>
+        /// <param name="args">The arguments passed to CreateDefaultBuilder, typically from <c>Program.Main(string [] args)</c>.</param>
+        /// <returns>An <see cref="IWebHostBuilder"/> instance.</returns>
+        public static IWebHostBuilder FromStartup<TStartup>(string[] args)
```

Comment:
This isn't an extension, can we find a new home for it?

Reply:
I'm moving these to TestServer after I talk to Chris